repo/actions/registry.go

```diff
@@ -42,6 +42,21 @@ func (act Registry) Unpublish(ref repo.DatasetRef) (err error) {
 	return cli.DeleteDataset(ref.Peername, ref.Name, ds.Encode(), pub)
 }
 
+// Status checks to see if a dataset is published to a repo's specific registry
+func (act Registry) Status(ref repo.DatasetRef) (err error) {
```

Comment:
I like it. In the near future registries are going to get an upgrade where they can also pin, and we'll have to modify this func to return the status of two things:
* if a given ref is published to the registry (what it's currently doing)
* if that given ref is _pinned_ to the registry (commng soon)

For now this is a great starting point.